SConstruct

```diff
@@ -106,23 +106,15 @@ linker_to_flags = {
   'clang++'  : clang_linker_flags
 }
 
-
-def cuda_installation():
+def cuda_installation(env):
```

Comment:
omp_installation is not a good example, IMO. All it's used for is to figure out library name which it does derive from compiler name.
In this case I can either pass cuda_path instead of env or pass nothing, return paths relative to CUDA installation and then stitch them into absolute path where it's used. I don't think either of these options improves things.